waspy/app.py

```diff
@@ -161,9 +175,18 @@ def run(self):
                 handler = self.router.get_handler_for_request(request)
                 request.app = self
                 response = await handler(request)
-
+                response.app = self
             except ResponseError as r:
-                response = r.response
+                parser = app_parsers.get(request.content_type, None)
+                if not parser:
+                    content_type = self.default_content_type
+                else:
+                    content_type = request.content_type
```

Comment:
Should the response content_type be read from the parser.content_type instead of the request?

Reply:
It is essentially the same. I am using the request here because I was able to grab a parser matching it.